src/main/java/com/snowflake/kafka/connector/internal/streaming/OpenChannelRetryPolicy.java

```diff
@@ -38,59 +38,36 @@ class OpenChannelRetryPolicy {
   /**
    * Executes the provided channel opening action with retry handling.
    *
-   * <p>On streaming-related exceptions, it will retry with exponential backoff and jitter. If all
-   * retries are exhausted, it will throw the last encountered exception.
+   * <p>On SFException containing "429" (HTTP rate limiting), it will retry with exponential backoff
+   * and jitter with unlimited retry attempts. Other exceptions are not retried.
    *
    * @param channelOpenAction the action to execute (typically openChannelForTable call)
    * @param channelName the channel name for logging purposes
-   * @param connectorConfig the connector configuration map
    * @return the result of the channel opening operation
    */
   static SnowflakeStreamingIngestChannel executeWithRetry(
-      CheckedSupplier<SnowflakeStreamingIngestChannel> channelOpenAction,
-      String channelName,
-      Map<String, String> connectorConfig) {
-
-    // Get the configurable max delay, using default if not specified
-    int maxRetryAttempts =
-        Integer.parseInt(
-            connectorConfig.getOrDefault(
-                SnowflakeSinkConnectorConfig.OPEN_CHANNEL_MAX_RETRY_ATTEMPTS,
-                String.valueOf(
-                    SnowflakeSinkConnectorConfig.OPEN_CHANNEL_MAX_RETRY_ATTEMPTS_DEFAULT)));
-
-    Fallback<SnowflakeStreamingIngestChannel> fallback =
-        Fallback.ofException(
-            e -> {
-              LOGGER.error(
-                  "Open channel {} - max retry attempts reached. Last exception: {}",
-                  channelName,
-                  e.getLastException().getMessage(),
-                  e.getLastException());
-              throw e.getLastException();
-            });
+      CheckedSupplier<SnowflakeStreamingIngestChannel> channelOpenAction, String channelName) {
 
     RetryPolicy<SnowflakeStreamingIngestChannel> retryPolicy =
         RetryPolicy.<SnowflakeStreamingIngestChannel>builder()
-            .handle(SFException.class)
+            .handleIf(e -> e instanceof SFException && shouldBeRetried(e))
```

Comment:
i would just call whole statement as shouldBeRetried
```suggestion
            .handleIf(OpenChannelRetryPolicy::shouldBeRetried)
```